### DISCUSSION TITLE
React Reference Type

### DISCUSSION THREAD

**Discussion by kwburnett** [ericgio/react-bootstrap-typeahead#800] [Q&A] [answered]
Thanks for creating a great and helpful library! I'm trying to upgrade from 4.x to the latest (currently 6.2.3) and have a question about what type to use when creating my own reference. Using the example from [Methods.md](https://github.com/ericgio/react-bootstrap-typeahead/blob/main/docs/Methods.md#public-methods), what type should the reference have?
```javascript
const ref = React.createRef<~>();
                            ^ what type here?
<>
  <Typeahead
    ...
    ref={ref}
  />
  <button onClick={() => ref.current.clear()}>
    Clear Typeahead
  </button>
</>
```

I've also [created a sandbox](https://codesandbox.io/s/modest-chaum-sxsy87?file=/src/App.tsx), if anyone wants to mess around (the most recent version of `react-bootstrap-typeahead` is 6.0.2 there). In my own testing with 6.2.3, I get the following when trying to use `Typeahead` exported from the library:
![image](https://github.com/ericgio/react-bootstrap-typeahead/assets/1461567/a01a3a00-d220-4347-93b5-f7d502d9d61a)

or, if I update it to be `useRef<typeof Typeahead | null`, then I'll get the following error when trying to specify the methods:
![image](https://github.com/ericgio/react-bootstrap-typeahead/assets/1461567/91aebb6b-b29c-4487-887d-e6b2d8f9e118)

Any help would be greatly appreciated. Regardless, thanks again for this sweet library!

**Comment by ericgio** [comment, marked as answer]
Hey @kwburnett, thanks for the question! As of v6.2, you can import `TypeaheadRef` from the package and use that to type the ref:

```tsx
import { Typeahead, TypeaheadRef } from 'react-bootstrap-typeahead';

...

const ref = useRef<TypeaheadRef>();
```

**Reply by kwburnett** [reply]
That's exactly what I needed - thank you!
